kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt

```diff
@@ -70,22 +76,61 @@ internal class WorkflowNode<PropsT, StateT, OutputT : Any, RenderingT>(
    */
   override val coroutineContext = baseContext + Job(baseContext[Job]) + CoroutineName(id.toString())
 
-  private val subtreeManager = SubtreeManager<StateT, OutputT>(coroutineContext)
+  /**
+   * ID used to uniquely identify this node to [WorkflowDiagnosticListener]s.
+   *
+   * Visible for testing.
+   */
+  internal val debugId = hashCode().toLong()
+
+  private val subtreeManager =
+    SubtreeManager<StateT, OutputT>(coroutineContext, debugId, diagnosticListener)
+
   private val workerTracker =
     LifetimeTracker<WorkerCase<*, StateT, OutputT>, Any, WorkerSession>(
         getKey = { case -> case },
-        start = { case -> WorkerSession(launchWorker(case.worker)) },
+        start = { case ->
+          val debugIdReservation = Any()
```

Comment:
This is kind of gross, need to find a better way to generate the ID. Maybe just a monotonically-increasing property in the node, combined with the node's ID.

Reply:
Yeah, this worries me. Seems okay to merge and lock in gains, but should track.

Reply:
I'm gonna fix it first, this is some sloppy 1am shit.